### ISSUE TITLE
[Good First Issue]: Add basic git based code checks

### ISSUE THREAD

**Issue by Wovchena** [openvinotoolkit/openvino.genai#663]
### Context
As a first step to improving GenAI code style, basic checks defined in the project's pre-commit configuration should be run.

### What needs to be done?
1. Run 3 checks from https://github.com/openvinotoolkit/model_api/blob/66384b0b747dee0fb3e37c6466115b7e60da3a10/.github/workflows/test_precommit.yml#L12-L16. Configure `pre-commit` python module to also `check-merge-conflict` and `check-json`. Python (e.g., `black --check .`) and C++ specific checks are out of scope. A good example is in https://github.com/openvinotoolkit/openvino.genai/pull/1453: https://github.com/openvinotoolkit/openvino.genai/blob/33f87c15980ef4296a7dd1f7775c27f622db8f86/.pre-commit-config.yaml.
2. Fix detected issues. Some files are expected to fail. For example ™ is an expected non ASCII symbol in root README.md. `src/python/openvino_genai/py_openvino_genai.pyi` should also be excluded because it's an automatically generated file. It should be regenerated by compiling the project and added to your commit. Exclude such files from testing.
3. Add a new job within the GitHub Actions workflow to https://github.com/openvinotoolkit/openvino.genai/blob/master/.github/workflows/causal_lm_cpp.yml to run these three checks.
4. Submit a pull request (PR).

### Example Pull Request
https://github.com/openvinotoolkit/openvino.genai/pull/315

### Resources
- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers

### Contact points
@Wovchena

**Comment by tejasathalye** [comment]
.take

**Comment by Wovchena** [comment]
@LitZeus, are you still working on this?

**Comment by LitZeus** [comment]
no, I am not getting the right solution...
@Wovchena

**Comment by Wovchena** [comment]
Can you describe the problem?

**Comment by Wovchena** [comment]
Feel free to take the issue again if you work on it.

**Comment by rishik-ashili** [comment]
.take

**Comment by Wovchena** [comment]
@rishik-ashili, do you still work on this?

**Comment by rishik-ashili** [comment]
Yes I'm trying

**Comment by Wovchena** [comment]
@rishik-ashili, do you still work on this?

**Comment by Cyber-Var** [comment]
.take

**Comment by Cyber-Var** [comment]
Sure, thank you!

**Comment by FReakYdiVi** [comment]
.take

**Comment by FReakYdiVi** [comment]
@Cyber-Var do you still working on this issue?

**Comment by Cyber-Var** [comment]
Yes, I am

**Comment by Cyber-Var** [comment]
I've created a pull-request.
https://github.com/openvinotoolkit/openvino.genai/pull/1453

**Comment by harshalubale4** [comment]
Hey @p-wysocki, can I work on this issue if this is not closed...?

**Comment by Wovchena** [comment]
> Hey [@p-wysocki](https://github.com/p-wysocki) and [@Wovchena](https://github.com/Wovchena) , can I work on this issue if this is not closed...?

Let's wait until Monday if @Cyber-Var replyes.

@Cyber-Var, do you still working on this issue?

**Comment by Cyber-Var** [comment]
@Wovchena , yes, I am still working on it. I made updates on my pull request 2 weeks ago and am waiting for review there: https://github.com/openvinotoolkit/openvino.genai/pull/1453

**Comment by Wovchena** [comment]
You need to fix pre commit builds

**Comment by Wovchena** [comment]
@Cyber-Var, do still work on this?

**Comment by thesakshidiggikar** [comment]
.take

**Comment by thesakshidiggikar** [comment]
@Wovchena can you check and have review?

**Comment by Wovchena** [comment]
I reviewed. The result is https://github.com/openvinotoolkit/openvino.genai/pull/1832#discussion_r1977088962

**Comment by thesakshidiggikar** [comment]
i want bit help from you @Wovchena

**Comment by Wovchena** [comment]
@thesakshidiggikar, do still work on this?

**Comment by Raghavyadav17** [comment]
.take

**Comment by mahimairaja** [comment]
Hey @Wovchena is this issue stale?

**Comment by Wovchena** [comment]
No. Nobody just picks it up

**Comment by mahimairaja** [comment]
Got it! @Wovchena can you please assign me this issue, I am interested to adding the Github Actions workflow for the defined task

For clarity, we would run this pre commit check only when a PR is raised against `master` branch right? before merging

**Comment by Wovchena** [comment]
`master` and `releases/**`. You can see the trigger condition at the top of `.github/workflows/linux.yml`

**Comment by goyaladitya05** [comment]
.take

**Comment by Wovchena** [comment]
I suspect @sgonorov just implemented it. @sgonorov, can you confirm?

**Comment by Wovchena** [comment]
Yes

**Comment by sgonorov** [comment]
Yes, it's implemented on a base level. Will be working on expanding linting/formatting functionality in the future.

**Comment by amishijain123** [comment]
Hello OpenVino team 👋
I’m a beginner contributor and would love to work on this issue.
Is it okay for me to take this up?
Thank you!

**Comment by Wovchena** [comment]
No. It's mostly implemented by @sgonorov 

I thought I removed this issue from the list of good first issues. How did you find it?

**Comment by amishijain123** [comment]
.take

**Comment by Aryansinghrathore7** [comment]
Add basic pre-commit code quality checks

This PR introduces basic pre-commit checks as a first step toward improving
GenAI code style consistency.

Changes included:
- Added pre-commit configuration with basic checks:
  - check-added-large-files
  - check-ast
  - check-yaml
  - check-merge-conflict
  - check-json
- Excluded expected failing and auto-generated files
- Fixed issues detected by the new checks
- Added a GitHub Actions job to run pre-commit checks in CI

Python formatting and C++ checks are intentionally out of scope.
 
pre-commit-config.yaml

repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.5.0
    hooks:
      - id: check-added-large-files
      - id: check-ast
      - id: check-yaml
      - id: check-merge-conflict
      - id: check-json
        exclude: |
          src/python/openvino_genai/py_openvino_genai\.pyi

exclude: |
  README\.md

**Comment by Palakb273** [comment]
.take

**Comment by Palakb273** [comment]
@sgonorov would you please let me know the status can i take this pr

**Comment by goyaladitya05** [comment]
> No. It's mostly implemented by [@sgonorov](https://github.com/sgonorov)
> 
> I thought I removed this issue from the list of good first issues. How did you find it?

@Wovchena It still appears under the GenAI section of the good first issues board , so contributers can find it from there.

https://github.com/orgs/openvinotoolkit/projects/3
